helpers.ts

```diff
@@ -388,6 +388,49 @@ export async function connectEventuallyUntilTimeout(factory: () => net.Socket, t
 	});
 }
 
+export function isPortAvailable(candidatePort: number): Promise<boolean> {
+	return new Promise<boolean>((resolve, reject) => {
+		let isResolved = false;
+		let server = net.createServer();
+
+		server.on("error", (err: Error) => {
+			if (!isResolved) {
+				isResolved = true;
+				resolve(false);
+			}
+		});
+
+		server.once("close", () => {
+			if (!isResolved) { // "close" will be emitted right after "error"
+				isResolved = true;
+				resolve(true);
+			}
+		});
+
+		server.on("listening", (err: Error) => {
+			if (err && !isResolved) {
+				isResolved = true;
+				resolve(false);
+			}
+
+			server.close();
+		});
+
+		server.listen(candidatePort, "localhost");
+	});
+}
+
+export async function getAvailablePort(port: number): Promise<number> {
```

Comment:
the correct way to get available port is by starting server on port 0 and waiting for it's listening event, where you'll receive the real free port. So I suggest either renaming the method or use 
https://github.com/telerik/mobile-cli-lib/blob/master/services/net-service.ts#L4